service/broadcast/src/main/java/org/kaazing/gateway/service/broadcast/BroadcastServiceHandler.java

```diff
@@ -59,6 +63,13 @@ IoHandler getListenHandler() {
     }
 
     @Override
+    public void messageReceived(IoSession session, Object message) throws Exception {
+        if (onClientMessage == BROADCAST) {
+            handler.messageReceived(session, message);
```

Comment:
Two questions:
- It broadcasts to all the clients (including itself)
- Do you see any thread alignment issues ?